src/typed-ast/ast.h

```diff
@@ -0,0 +1,160 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "typed-ast/expr.h"
+#include "typed-ast/node.h"
+
+/// This directory contains a typed AST representation of Verona programs.
+///
+/// "typed" in this context refers to the C++ classes that structure the AST,
+/// not about the contents of the AST. This is in constract with peglib's
+/// untyped AST, which are effectively S-expressions, with no well-defined
+/// structure.
+///
+/// In the long run, we aim to replace most/all uses of the untyped AST with the
+/// typed one.
+///
+/// The AST supports [LLVM-style RTTI]. Nodes that have multiple variants (eg.
+/// expressions) declare a `Kind` enum, with a value per concrete node. The kind
+/// is exposed through a `getKind` method and a static `classof` method. These
+/// are generally not used directly, but through LLVM's `isa<>`, `dyn_cast<>`,
+/// `TypeSwitch`, ... utilities.
+///
+/// The AST currently uses one allocation per node, and its lifetime is managed
+/// automatically using `std::unique_ptr` on every edge of the tree. In the
+/// future, we should consider using an arena to allocate all the nodes, since
+/// we are unlikely to ever deallocate individual nodes.
+///
+/// [LLVM-style RTTI]: https://llvm.org/docs/HowToSetUpLLVMStyleRTTI.html
+namespace verona::ast
+{
+  /// This is the base class for entity members, from which all concrete nodes
+  /// derive.
+  class MemberDef : public Node
```

Comment:
What's behind the name `MemberDef`? Why not `NodeDef`? To me, `Member` means a class member (field, method).

Reply:
It is a class member (well an entity member, which could be a class/module/interface). Why would it be `NodeDef`?

Reply:
Hm, I see, "Entity member definition". The way I read it (and its usage in field access below) makes me think of "class member". I think if we consider modules, interfaces and classes as the same thing, than this is actually correct.

Perhaps a more elaborate comment on that before the declaration would help those uninitiated in Verona parlay. :)